### PR TITLE
Timepicker - minute increment (ios/android)

### PR DESCRIPTION
**Pull request by AlexTrepanier**
## PR Type
Added time increment to ios and android

## PR Checklist
- [x] Tested code on ios, android and uwp
- [x] Docs have been added/updated [TimePicker.md](https://github.com/nventive/Uno/blob/master/src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePicker.md)
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
143859

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TimePicker/Sample1.xaml [hunk 1]
```diff
@@ -11,28 +11,21 @@
...
+			<DataTemplate>
+				<Grid Background="White">
+					<TimePicker Grid.Row="1"
                                 Margin="15,10"
                                 ios:FlyoutPlacement="Full"
                                 HorizontalAlignment="Center"
                                 VerticalAlignment="Top"
-                                Time="{Binding Time, Mode=TwoWay}"
-                                MinuteIncrement="1"
+                                MinuteIncrement="13"
```
Comment (MatFillion) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TimePicker/Sample1.xaml]: Shouldn't Sample1 have a minute increment of 1 or just no specifications at all ?  (be more like default)

Comment (MatFillion) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TimePicker/Sample1.xaml]: SampleDescription says 1

File: src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutBase.cs [hunk 2]
```diff
@@ -73,10 +73,7 @@ internal void Hide(bool canCancel)
 			Closed?.Invoke(this, EventArgs.Empty);
 		}
 
-		internal protected virtual void Close()
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutBase.cs]: Why is this necessary?

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePicker.md [hunk 3]
```diff
@@ -23,6 +27,7 @@ Native time picker is wrapped in the flyout.
 Timepicker flyout appear centered to screen.
 You can change the flyout button by copying and modifying TimePickerFlyoutButtonStyle.
 You can change the flyout button by copying and modifying TimePickerFlyoutPresenterStyle.
+When time picker use minute increment more than 1 minutes, his style default to "spinner mode"
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePicker.md]:
```suggestion
If `MinuteIncrement` is not 1, `TimePicker` will show in "spinner mode"
```

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs [hunk 4]
```diff
@@ -18,16 +18,15 @@ internal protected override void Open()
...
-			if (Build.VERSION.SdkInt == BuildVersionCodes.M &&
-			    Build.Manufacturer.ToLower().IndexOf("samsung") >= 0)
-			{
-				ShowUsingAlertDialog();
-			}
-			else
-			{
-				ShowUsingTimePickerDialog();
-			}
+			//Minute increments are not supported with clock and Samsung devices with Marshmellow
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs]:
```suggestion
			//Minute increments are not supported with clock and Samsung devices with marshmallow (Android 6.0)
```

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs [hunk 5]
```diff
@@ -42,65 +41,44 @@ private void SaveTime(int hourOfDay, int minutes)
...
 
 		private void OnDismiss(object sender, EventArgs e)
 		{
 			Hide(canCancel: false);
 		}
 
-		partial void OnTimeChangedPartialNative(TimeSpan oldTime, TimeSpan newTime){}
+		partial void OnTimeChangedPartialNative(TimeSpan oldTime, TimeSpan newTime) { }
+
+		public class OnSetTimeListener : Java.Lang.Object, IOnTimeSetListener
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs]: Instead of creating a new class, could you implement IOnTimeSetListener on TimePickerFlyout directly?  (and pass `this` to the dialog).

Comment (MatFillion) [reply src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs]: I'm not entirely sure of the best practice. @jeromelaban

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+namespace Windows.UI.Xaml.Controls
+{
+	internal static class TimeSpanExtensions
+	{
+		internal static TimeSpan RoundToInterval(this TimeSpan time, int interval)
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs]: Add /// summary.

Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs]:
```suggestion
		internal static TimeSpan RoundToInterval(this TimeSpan time, int minuteInterval)
```
a way to clarify that the unit is minute.

Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs]: The description should specify the unit for the `interval` parameter.

File: src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutBase.cs (hunk 2 above)
Comment (AlexTrepanier) [reply src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutBase.cs]: need to be implemented in derived classes ..

I just removed the useless and dangerous 
throw new NotImplementedException("This method has no base class implementation and must be overridden in a derived class.");

since base.close() would crash

note that i havent cleaned Open() and CreatePresenter either.. but they should be cleaned..

flyoutbase should actually be abstract and these should be abstract method or virtual but empty. 

here where close() is used.. 
internal void Hide(bool canCancel)
		{
			if (!_isOpen)
			{
				return;
			}

			if (canCancel)
			{
				var closing = new FlyoutBaseClosingEventArgs();
				Closing?.Invoke(this, closing);
				if (closing.Cancel)
				{
					return;
				}
			}

			Close();
			_isOpen = false;
			Closed?.Invoke(this, EventArgs.Empty);
		}

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs (hunk 5 above)
Comment (AlexTrepanier) [reply src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs]: class OnSetTimeListener need to implement Java.Lang.Object interface so i think it's better as internal class like now. 

public class OnSetTimeListener : Java.Lang.Object, IOnTimeSetListener

File: src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs (hunk 6 above)
Comment (AlexTrepanier) [reply src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs]: summary to the class or to the method.?

i can add but i don't see why for both the class and the method the name is self explanatory. 

also i'm moving the method to Uno.UI.Extensions.TimeSpanExtensions

Comment (AlexTrepanier) [reply src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs]: yes good point, i'll rename the method to RoundToNextMinuteInterval

Comment (MatFillion) [reply src/Uno.UI/UI/Xaml/Controls/TimePicker/TimeSpanExtensions.cs]: on the method.
